set.mm

```diff
@@ -717119,6 +717125,529 @@ Graph theory (extension)
+
+  In the following, a general definition of the isomorphy relation for graphs
+  and specializations for simple hypergraphs ( ~ isomushgr ) and simple
+  pseudographs ( ~ isomuspgr ) are provided.  The latter corresponds to the
+  definition in [Bollobas] p. 3).  It is shown that the isomorphy relation for
+  graphs is an equivalence relation ( ~ isomgrref , ~ isomgrsym , ~ isomgrtr ).
+  Fianlly, isomorphic graphs with different representations are studied
+  ( ~strisomgrop , ~ ushrisomgr ).
+
+  Maybe more important than graph isomorphy is the notion of graph isomorphism,
+  which can be defined as in ~ df-grisom .  Then ` A IsomGr B <-> `
+  ` E. f f e. ( A GrIsom B ) ` resp. ` A IsomGr B <-> ( A GrIsom B ) =/= (/) `.
+  Notice that there can be multiple isomorphisms between two graphs (let
+  ` <. { A , B } , { { A , B } } >. ` and ` <. { { M , N } , { { M , N } } >. `
+  be two graphs with two vertices and one edge, then ` A |-> M , B |-> N ` and
+   ` A |-> N , B |-> M ` are two different isomorphisms between these graphs).
+
+  Another approach could be to define a category of graphs (there are maybe
+  multiple ones), where graph morphisms are couples consisting in a function on
+  vertices and a function on edges with required compatibilities, as used
+  in the definition of ` GrIsom `.  And then, a graph isomorphism is defined as
+  an isomorphism in the category of graphs (something like "GraphIsom
+  = ( Iso `` GraphCat )" ).  Then general category theory theorems could be
+  used, e.g., to show that graph isomorphy is an equivalence relation.
+
+$)
+
+  $( Introduce new constant symbols. $)
+  $c GrIsom $. $( Graph isomorphisms. $)
+  $c IsomGr $. $( Predicate "is isomorphic" for graphs $)
+
+  $( Extend class notation to include the graph ispmorphisms. $)
+  cgrisom $a class GrIsom $.
+
+  $( Extend class notation to include the isomorphy relation for graphs. $)
+  cisomgr $a class IsomGr $.
+
+  ${
+    $d f g i x y $.
+    $( Define the class of all isomorphisms between two graphs.  (Contributed
+       by AV, 11-Dec-2022.) $)
+    df-grisom $a |- GrIsom = ( x e. _V , y e. _V |-> { <. f , g >. |
+                       ( f : ( Vtx ` x ) -1-1-onto-> ( Vtx ` y )
+                      /\ g : dom ( iEdg ` x ) -1-1-onto-> dom ( iEdg ` y )
+                      /\ A. i e. dom ( iEdg ` x ) ( f " ( ( iEdg ` x ) ` i ) )
+                                        = ( ( iEdg ` y ) ` ( g ` i ) ) ) } ) $.
+  $}
+
+  ${
+    $d f g i x y $.
+    $( Define the isomorphy relation for graphs.  (Contributed by AV,
+       11-Nov-2022.) $)
+    df-isomgr $a |- IsomGr = { <. x , y >. |
+                          E. f ( f : ( Vtx ` x ) -1-1-onto-> ( Vtx ` y )
+            /\ E. g ( g : dom ( iEdg ` x ) -1-1-onto-> dom ( iEdg ` y )
+                      /\ A. i e. dom ( iEdg ` x ) ( f " ( ( iEdg ` x ) ` i ) )
+                                        = ( ( iEdg ` y ) ` ( g ` i ) ) ) ) } $.
```

Comment:
Now that you have df-grisom, there is no reason not to define
```
$a |- ( A IsomGr B <-> E. f f e. ( A GrIsom B ) ) $.
```
(?)

Reply:
Yes, this would be possible, but I would have to change/add some theorems, for which I do not want to spend time at the moment. Maybe in a later PR (at least when the material goes to main, if ever...).